File: functions/models/message.go
package models

import "time"

const NewUserMessage = "Dê as boas vindas ao usuário %s"

var AdminUser = User{
	Id:   "admin",
	Name: "ColaborAtiva",
}

type UserMessage struct {
	Content string `json:"content" validate:"required"`
	To      string `json:"to"`
}

type MessageEmitter struct {
	Id   string `json:"id" validate:"required"`
	Name string `json:"name" validate:"required"`
}

type Message struct {
	Timestamp time.Time      `json:"timestamp" validate:"required"`
	Content   string         `json:"content" validate:"required"`
	Emitter   MessageEmitter `json:"emitter" validate:"required"`
}

File: app/lib/modules/chat/core/models/message.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import 'emitter.dart';

part 'message.freezed.dart';

@freezed
class Message with _$Message {
  const Message._();

  const factory Message({
    required String id,
    required String content,
    required DateTime timestamp,
    required Emitter emitter,
  }) = _Message;
}
